### PR TITLE
Add jobs for the HCO

### PR DESCRIPTION
**Pull request by gbenhaim**
Signed-off-by: gbenhaim <galbh2@gmail.com>

### CODE REVIEW

File: github/ci/prow/files/config.yaml [hunk 1]
```diff
@@ -60,6 +60,7 @@ tide:
     kubevirt/kubectl-virt-plugin: squash
     kubevirt/node-maintenance-operator: squash
     kubevirt/ansible-kubevirt-modules: squash
+    kubevirt/hyperconverged-cluster-operator: merge
```
Comment (danielBelenky) [inline github/ci/prow/files/config.yaml]: Is it a request of the project to use `merge` instead of `squash` like everyone else?

Comment (rmohr) [reply github/ci/prow/files/config.yaml]: you can use whatever HCO maintainers prefer.

Comment (tiraboschi) [reply github/ci/prow/files/config.yaml]: no, on the project itself we are using squash

File: github/ci/prow/files/jobs/hyperconverged-cluster-operator/hyperconverged-cluster-operator-presubmits.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    spec:
+      nodeSelector:
+        type: bare-metal-external
+      containers:
+        - image: docker.io/kubevirtci/kubevirt-infra-bootstrap:v20191223-f0a2baa
+          command:
+            - "/usr/local/bin/runner.sh"
+            - "/bin/sh"
+            - "-c"
+            - "export TARGET=okd-4.1 && automation/test.sh"
```
[the remark this answers is not in the document]
Comment (tiraboschi) [reply github/ci/prow/files/jobs/hyperconverged-cluster-operator/hyperconverged-cluster-operator-presubmits.yaml]: pull-hyperconverged-cluster-operator-e2e-ocp-4.3 -> ocp-4.3

[the remark this answers is not in the document]
Comment (gbenhaim) [reply github/ci/prow/files/jobs/hyperconverged-cluster-operator/hyperconverged-cluster-operator-presubmits.yaml]: fixed

File: github/ci/prow/files/config.yaml (hunk 1 above)
Comment (gbenhaim) [reply github/ci/prow/files/config.yaml]: fixed

**Comment by danielBelenky** [comment]
/lgtm

@tiraboschi feel free to approve if you find this PR ok

**Comment by gbenhaim** [comment]
In order to fetch the okd images we need a secret, I've updated the job accordingly.

**Comment by gbenhaim** [comment]
```
+ /bin/sh -c 'cat $QUAY_PASSWORD | docker login --username $(cat $QUAY_USER) --password-stdin quay.io && export TARGET=ocp-4.3 && automation/test.sh'
Error: Cannot perform an interactive login from a non TTY device
```

@danielBelenky do you have any idea how to solve it? maybe we need to allocate a tty for the test container.

**Comment by gbenhaim** [comment]
I can't test the ocp lane, since `mkpj` doesn't take into consideration presets that were declared outside of `config.yaml` let's merge it and then test.

**Pull request merged**